hack/run-hive-locally.sh

```diff
@@ -0,0 +1,296 @@
+#!/bin/bash
+
+# TODO: hiveadmission support
+# TODO: run hive-controllers & hive-clustersync separately with multiple replicas each (need docker to avoid metrics port collision)
+
+set -e
+set -u
+set -o pipefail
+
+# Highlighted echo
+function hl_echo {
+    tput setaf 3
+    echo "$@"
+    tput sgr0
+}
+
+function usage {
+    echo "The script runs a Hive component ($hive_component_operator or $hive_component_controllers) locally,
+simulating in-cluster environment. It records the current in-cluster Hive configurations,
+scales down the selected component, then runs it locally with the same environment.
+On pressing ctrl + c, the script is stopped and relevant components are scaled back up.
+
+Notes:
+- Artifacts (in-cluster Hive configurations and log files), if produced, are stored in the --artifacts_folder
+which is removed after the script ends if the --cleanup_artifacts is specified.
+- Selecting $hive_component_controllers runs all Hive controllers (including hive-clustersync) together
+as a single replica.
+
+Usage:
+- ./hack/run-hive-locally.sh [--flag=value ...]
+
+Supported flags:
+--hive-operator-ns=NAMESPACE
+--hive_operator_log_level=LOG_LEVEL
+--hive_controllers_log_level=LOG_LEVEL
+--artifacts_folder=DIRECTORY
+--cleanup_artifacts
+--help"
+}
+
+function kill_background_jobs {
+    while read -r pid; do
+        hl_echo "Killing job $pid"
+        kill "$pid" || true
+    done < <(jobs -p)
+    sleep "$kill_job_grace_period_seconds"
+}
+
+function get_incluster_config {
+    hl_echo "Getting in-cluster Hive configurations"
+
+    resolved_env_values=()
+    local env_configmap_keys=(
+        "MANAGED_DOMAINS_FILE/hive-managed-domains/managed-domains" \
+        "AWS_PRIVATELINK_CONTROLLER_CONFIG_FILE/aws-private-link/aws-private-link" \
+        "FAILED_PROVISION_CONFIG_FILE/hive-failed-provision-config/hive-failed-provision-config" \
+        "METRICS_CONFIG_FILE/hive-metrics-config/hive-metrics-config"
+    )
+    local envfrom_configmaps=(
+        "hive-controllers-config"
+    )
+    local env
+    local value
+    local configmap
+
+    # Creates artifacts folder if non-existent
+    if ! [[ -d "./$artifacts_folder" ]]; then
+        hl_echo "Artifacts folder $artifacts_folder not found, creating"
+        mkdir "$artifacts_folder"
+    fi
+
+    # Resolve some necessary env's
+    local key
+    for env_configmap_key in "${env_configmap_keys[@]}"; do
+        env="${env_configmap_key%%/*}"
+        configmap="$(echo "$env_configmap_key" | cut -d '/' -f 2)"
+        key="${env_configmap_key##*/}"
```

Comment:
Great to know, thanks. The current version makes use of associative arrays to avoid parsing environment variables / ConfigMaps that are already read. 

(This also makes me upgrade Bash on my mac :joy:. )